Duplicati/Library/Main/Backend/BackendManager.Handler.cs

```diff
@@ -234,16 +275,23 @@ private async Task Run(IReadChannel<PendingOperationBase> requestChannel)
                         // Allow PUT operations to be queued, if requested
                         if (op is PutOperation putOp && !putOp.WaitForComplete)
                         {
-                            await EnsureAtMostNPendingUploads(maxParallelUploads).ConfigureAwait(false);
+                            await EnsureAtMostNActiveUploads(maxParallelUploads).ConfigureAwait(false);
 
                             // Operation is accepted into queue, so we can signal completion
                             putOp.SetComplete(true);
                             activeUploads.Add(ExecuteWithRetry(putOp, tcs.Token));
                         }
+                        else if (op is GetOperation getOp && !getOp.WaitForComplete)
+                        {
+                            await EnsureAtMostNActiveDownloads(maxParallelDownloads).ConfigureAwait(false);
+
+                            // Operation is accepted into queue, so we can signal completion
+                            activeDownloads.Add(ExecuteWithRetry(getOp, tcs.Token));
+                        }
                         else
                         {
-                            // Wait for all pending uploads to complete
-                            await EnsureAtMostNPendingUploads(1).ConfigureAwait(false);
+                            // Wait for all of the active uploads to complete
+                            await EnsureAtMostNActiveUploads(1).ConfigureAwait(false);
```

Comment:
Do we also need to ensure that there are no pending downloads at this stage? 
The way the code is constructed this will not currently happen, but maybe it is logical that we have flushed active operations when we do something "synchronous" ?